### ISSUE TITLE
equivalent of tensorflow.nn.softmax() in theano

### ISSUE THREAD

**Issue by saitarslanboun** [Theano/Theano#6536]
Hi,

In tensorflow, a 5D tensor is given to tensorflow.nn.softmax(input, dim=2) function.

I would like to do the same with theano, and I am very new. Could you help me to create an equivalent function in theano?

Thanks,

**Comment by nouiz** [comment]
I'm not sure what tensorflow does in that case. Can you reshape/dimshuffle your 5d tensor to 2d and call theano.tensor.nnet.softmax? We where working to have a better interface to support all softmax case, but it isn't done.

dimshuffle is our n-dim transpose operator. Move the dimensions you want to softmax over on the last dimensions and collapse them in 1 dimensions. Explained otherwise, our softmax take 2 dimensions as input. The first dimensions is just the batch. So we do one softmax for each element in that dimensions.
